### ISSUE TITLE
steam do not starts at Ubuntu 22.04 (bwrap: Can't bind mount /oldroot/etc/drirc on /newroot/etc/drirc: Unable to mount source on destination: Permission denied)

### ISSUE THREAD

**Issue by Sfinx** [canonical/steam-snap#386]
### Ensure there isn't an existing issue for this and check the wiki

- [X] This issue is not a duplicate and I have checked the wiki.

### Current Behavior

steam installed from snap prints:

```
steamwebhelper, a critical Steam component, is not responding. The Steam UI will not be usable.  Click here for more information:
```

the ~/snap/steam/common/.local/share/Steam/logs/steamwebhelper.log contains:

```
steamwebhelper.sh[1209070]: === Wed May 29 00:32:47 EEST 2024 ===
steamwebhelper.sh[1209070]: Starting steamwebhelper under bootstrap sniper steam runtime at /home/rus/snap/steam/common/.local/share/Steam/ubuntu12_64/steam-runtime-sniper
pressure-vessel-wrap[1209070]: W: Failed to load Vulkan ICD #0 from /var/lib/snapd/lib/vulkan/icd.d/nvidia_icd.json: openat(/var/lib/snapd/lib/vulkan/icd.d/nvidia_icd.json): No such file or directory
pressure-vessel-wrap[1209070]: W: "snap/steam/192/graphics/usr/share/libdrm" is unlikely to appear in "/run/host"
pressure-vessel-wrap[1209070]: W: "snap/steam/192/graphics/usr/share/libdrm" is unlikely to appear in "/run/host"
pressure-vessel-wrap[1209070]: W: "snap/steam/192/graphics/usr/share/drirc.d" is unlikely to appear in "/run/host"
pressure-vessel-wrap[1209070]: W: Not sharing /snap with container to work around https://github.com/canonical/steam-snap/issues/359
pressure-vessel-wrap[1209070]: W: Not sharing /var/lib/snapd/hostfs with container to work around https://github.com/canonical/steam-snap/issues/359
pressure-vessel-wrap[1209070]: W: Filtering out --tmpfs /snap/steam/192/usr/share/vulkan/implicit_layer.d to work around https://github.com/canonical/steam-snap/issues/370
bwrap: Can't bind mount /oldroot/etc/drirc on /newroot/etc/drirc: Unable to mount source on destination: Permission denied
```

### Expected Behavior

steam should start

### Steps To Reproduce

```
snap install steam
steam
```

### Environment

```yaml
os_release:
    name:               "Ubuntu"
    version:            "22.04.4 LTS (Jammy Jellyfish)"
snap_info:
    steam_revision:     192
    snapd_revision:     21759
lspci:
glxinfo:
    gpu:                AMD Radeon Graphics (radeonsi, navi33, LLVM 15.0.7, DRM 3.57, 6.8.8-060808-generic)
    gpu_version:        4.6 (Core Profile) Mesa 24.0.3 - kisak-mesa PPA
lscpu:
    model_name:         AMD Ryzen 7 7840HS w/ Radeon 780M Graphics
xdg_current_desktop:    ubuntu:GNOME
desktop_session:        ubuntu
```

**Comment by Sfinx** [comment]
Adding the following to /var/lib/snapd/apparmor/profiles/snap.steam.steam fix the bug for me:

```
mount options=(rw, rbind) /oldroot/etc/drirc -> /newroot/etc/drirc,
```

**Comment by ashuntu** [comment]
We are reworking the app armor profiles (in [snapd](https://github.com/snapcore/snapd)) for the Steam snap pretty heavily this cycle, so at the moment this will remain unresolved until that work is complete, but should be resolved with those changes.

**Comment by zyga** [comment]
This has been fixed in snapd, and will be released in snapd 2.64.

You can give it a try with:
```
sudo snap refresh snapd --edge
```

To go back to a stable release of snapd please run:
```
sudo snap refresh snapd --stable
```

Some changes may need a reboot to apply (not really but easier than precise description).
